### ISSUE TITLE
Table - colgroup / col width not working

### ISSUE THREAD

**Issue by Nuranto** [dompdf/dompdf#3029]
Hi there,

I use tinymce to generate the html processed by dompdf. I guess we are a lot to do that.

I just updated to tinymce 6, and they changed the way of managing column width. The width was set on `td` before, with v6, the width are set in `colgroup>col` element, and that seem to be not managed by dompdf.

Here is an html example : 

```
<table style="border-collapse: collapse; width: 72.8395%; height: 58.7814px;" border="1">
        <colgroup>
            <col style="width: 76.3136%;">
            <col style="width: 23.6864%;">
        </colgroup>
        <tbody>
            <tr>
                <td>test</td>
                <td>test</td>
            </tr>
        </tbody>
</table>
```

**Expected result :** 

Width respected :

![Capture d’écran 2022-10-18 à 12 47 37](https://user-images.githubusercontent.com/1633463/196409757-193361c9-2002-47a9-903b-e5e562be6af4.png)

**Actual result :** 

50/50 width :

![Capture d’écran 2022-10-18 à 12 48 45](https://user-images.githubusercontent.com/1633463/196409943-913aad00-5315-4f9d-a6fe-e24b1ce8c4f6.png)

**Comment by ArnaudAploon** [comment]
Yes indeed tinymce to change its way of assigning widths. Also, <colgroup> is not supported by dompdf at all.
So instead of using <colgroup>, you can assign widths directly to the <td> elements in the first row of your <table>, like this (You can also use :nth-child)

```
         <style>

              .col-1 {
                width: 2.8%;
              }

              .col-2 {
                width: 53.8%;
              }

              .col-3 {
                width: 0.6%;
              }

              .col-4 {
                width: 4.2%;
              }             
              
         </style>
        
        <table style="table-layout: fixed; width: 100%;">
            <tbody>
                <tr style="">
                    <td class="col-1" style=""></td>
                    <td class="col-2" style=""></td>
                    <td class="col-3" style=""></td>
                    <td class="col-4" style=""></td>
                    <td class="col-5" style=""></td>
                </tr>
		...
```
On the other hand, there is also wkhtmltopdf (snappy) which is a very good alternative to dompdf. wkhtmltopdf solves your problem, since it uses WebKit rendering

**Comment by studioscully** [comment]
Just stumbled across this after having the same issue after upgrading to TinyMCE 6.

Good news: TinyMCE (at least at 6.8.3) allows you to turn off the colgroup method of table sizing.

To do that, set the "table_use_colgroups" parameter to **false** (it's true by default):

```
tinymce.init({
  selector: 'textarea',  // change this value according to your HTML
  table_use_colgroups: false
});

```
With that parameter set, TinyMCE uses the old method of per-td width info, and DomPDF is happy :)

More info at: https://www.tiny.cloud/docs/tinymce/latest/table-options/#table_use_colgroups
